Content.Server/Actions/ActionOnInteractSystem.cs

```diff
@@ -151,16 +173,29 @@ private List<Entity<ActionComponent, T>> GetValidActions<T>(List<EntityUid>? act
         return valid;
     }
 
-    private bool TryUseCharge(Entity<ActionOnInteractComponent> ent)
+    private bool EmptyCharges(Entity<ActionOnInteractComponent> ent)
     {
         if (!ent.Comp.RequiresCharge)
-            return true;
+        {
+            return false;
+        }
+
+        Entity<LimitedChargesComponent?> charges = ent.Owner;
+
+        return _charges.IsEmpty(charges);
+    }
```

Comment:
iirc you don't need to set to limited charges comp and can just pass ent.owner because it'll do the resolve.